syntax = "proto3";

package stream;

message StreamMessage {
  int64 seq_number = 1;
  string timestamp = 2;
  string payload = 3;
}

service StreamService {
  // Bidirectional streaming
  rpc StreamMessages (stream StreamMessage) returns (stream StreamMessage);
}